Выполненное третье домашнее задание.

gMarioboss opened the pull request.

gMarioboss committed:
Выполненное третье домашнее задание.

DwarfPorter commented on Lesson_2/Calculator/app/src/main/java/com/example/calculator/MainActivity.java:
```diff
@@ -3,12 +3,200 @@
+
+    private void initButtons() {
+        findViewById(R.id.button_0).setOnClickListener(v -> {
+            addToOperand("0");
+            makeHistory("0");
```
Отправляем "0" в константы

DwarfPorter commented on Lesson_2/Calculator/app/src/main/java/com/example/calculator/MainActivity.java:
```diff
@@ -3,12 +3,200 @@
+            makeHistory("0");
+        });
+        findViewById(R.id.button_1).setOnClickListener(v -> {
+            addToOperand("1");
+            makeHistory("1");
```
и "1" тоже в константы, и все остальные строки

DwarfPorter commented on Lesson_2/Calculator/app/src/main/java/com/example/calculator/MainActivity.java:
```diff
@@ -3,12 +3,200 @@
+            makeHistory("1");
+        });
+        findViewById(R.id.button_2).setOnClickListener(v -> {
+            addToOperand("2");
+            makeHistory("2");
```
Можно объединить и сделать метод, который эту пару вызывает

DwarfPorter commented on Lesson_2/Calculator/app/src/main/java/com/example/calculator/MainActivity.java:
```diff
@@ -3,12 +3,200 @@
+            makeHistory(".");
+        });
+        findViewById(R.id.button_plus).setOnClickListener(v -> {
+            result();
+            data.setOperationType('+');
+            makeHistory("+");
```
Тоже можно сделать отдельные метод

DwarfPorter reviewed (commented):
![image](https://user-images.githubusercontent.com/31162829/120906327-3c41ba80-c661-11eb-9da2-553783ae1126.png)

DwarfPorter commented:
дебажить надо.
Заметил, что после поворота экрана screenText обнулился, и выскочил exception
![image](https://user-images.githubusercontent.com/31162829/120906669-03efab80-c664-11eb-8f29-3809feb9b5ed.png)

gMarioboss committed:
HM3/CHANGES

Переписал немного логику, тем самым исправив множество багов. Исправил причину не сохранения состояния при Restore
